### PR TITLE
Add streaming metrics

### PR DESCRIPTION
**Pull request by khorshuheng**
**What this PR does / why we need it**:
Currently, we don't have any metrics to help us to determine if a streaming ingestion job can keep up with the incoming throughput. This PR add metrics for incoming and outgoing throughput.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
Additional metrics for stream ingestion.
```

### CODE REVIEW

File: spark/ingestion/src/main/scala/feast/ingestion/metrics/StreamingMetrics.scala [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+import org.apache.spark.SparkEnv
+import org.apache.spark.metrics.source.StreamingMetricSource
+import org.apache.spark.sql.streaming.StreamingQueryProgress
+
+class StreamingMetrics extends Serializable {
+
+  private lazy val metricSource: Option[StreamingMetricSource] = {
+    val metricsSystem = SparkEnv.get.metricsSystem
+    StreamingMetricsLock.synchronized {
```
Comment (pyalex) [inline spark/ingestion/src/main/scala/feast/ingestion/metrics/StreamingMetrics.scala:27]: no need to do lazy initialization, if metric are used on driver only

Comment (pyalex) [reply spark/ingestion/src/main/scala/feast/ingestion/metrics/StreamingMetrics.scala:27]: You can simply register source through spark config (see https://spark.apache.org/docs/latest/monitoring.html#metrics)

File: spark/ingestion/src/main/scala/feast/ingestion/StreamingPipeline.scala [hunk 2]
```diff
@@ -124,6 +148,19 @@ object StreamingPipeline extends BasePipeline with Serializable {
           .option("entity_repartition", "false")
           .save()
 
+        config.source match {
+          case _: KafkaSource =>
+            Option(
+              rowsAfterValidation
+                .agg(max("timestamp") as "latest_timestamp")
+                .collect()(0)
```
Comment (pyalex) [inline spark/ingestion/src/main/scala/feast/ingestion/StreamingPipeline.scala:156]: what if rowsAfterValidation would be empty?

Comment (khorshuheng) [reply spark/ingestion/src/main/scala/feast/ingestion/StreamingPipeline.scala]: Good catch, added validation

**Pull request merged**